### PR TITLE
Add a handle to the error and sending the request

### PR DESCRIPTION
**Pull request by maryna-velychko**

### CODE REVIEW

File: src/app/api.mjs [hunk 1]
```diff
@@ -184,3 +183,11 @@ export async function getSchools() {
   const response = await client.get(`${SCHOOLS_PATH}/`);
   return response.data;
 }
+
+
+// SIGNUP
+export async function signupStep1(data) {
```
Comment (shuryhin-oleksandr) [inline src/app/api.mjs:189]: Name it according to a component name

File: src/app/api.mjs [hunk 2]
```diff
@@ -184,3 +183,11 @@ export async function getSchools() {
...
+  const url = `${SIGNUP_PATH}/`;
+  const response = await client.post(url, data);
+  return response.data;
+}
```
Comment (shuryhin-oleksandr) [inline src/app/api.mjs:193]: No  empty line in the end f the file

File: src/app/signup/components/SignUpAccount.js [hunk 3]
```diff
@@ -1,13 +1,15 @@
...
-import { FormikCheckboxField, FormikTextField } from "@/app/activities/[activityId]/edit/components/formikFields";
+import { FormikCheckboxField, FormikTextField, FormikErrors, createHandleSubmit } from "@/app/activities/[activityId]/edit/components/formikFields";
 import HighlightOffRoundedIcon from "@mui/icons-material/HighlightOffRounded";
 import { Box, Button, Container, IconButton, Typography } from "@mui/material";
 import { Form, Formik } from "formik";
 import Image from "next/image";
 import { useRouter } from "next/navigation";
 import * as Yup from "yup";
+import { useMutation } from "react-query";
+import { signupStep1 } from "../../api.mjs";
```
Comment (shuryhin-oleksandr) [inline src/app/signup/components/SignUpAccount.js:12]: Use @/... paths

File: src/app/signup/components/SignUpAccount.js [hunk 4]
```diff
@@ -27,6 +29,14 @@ export function SignUpContainer({ children, sx }) {
 }
 
 export default function SignUpAccount({ goToNextStep }) {
+
+  const mutation = useMutation((value) => signupStep1({ email: value.email, password1: value.password1, password2: value.password2 }));
```
Comment (shuryhin-oleksandr) [inline src/app/signup/components/SignUpAccount.js:33]: Why not useMutation(signupStep1)?

File: src/app/signup/components/SignUpAccount.js [hunk 5]
```diff
@@ -27,6 +29,14 @@ export function SignUpContainer({ children, sx }) {
...
+
+  async function handleSubmit(values, formikHelpers) {
+    const handle = createHandleSubmit({ mutation, onSuccess: goToNextStep, throwError: true });
```
Comment (shuryhin-oleksandr) [inline src/app/signup/components/SignUpAccount.js:36]: Why throwError true?

File: src/app/signup/components/SignUpAccount.js (hunk 4 above)
Comment (maryna-velychko) [reply src/app/signup/components/SignUpAccount.js:33]: in this case, we will send termsConditionsAccepted too, but, I think, that doesn't matter

File: src/app/signup/components/SignUpAccount.js (hunk 5 above)
Comment (maryna-velychko) [reply src/app/signup/components/SignUpAccount.js:36]: if we don't need throw error I can delete it

**Pull request merged**